views/_layout.njk

```diff
@@ -0,0 +1,97 @@
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+    <!-- description block used to provider meta tag to describe page -->
+    {% block description %}{% endblock %}
+    <meta name="author" content="PSU Capstone F2017">
+    <!-- title block to list title of page to be displayed in browser tab -->
+    {% block title %}{% endblock %}
+
+    <!-- Bootstrap core CSS -->
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.3/css/bootstrap.min.css" integrity="sha384-Zug+QiDoJOrZ5t4lssLdxGhVrurbmBWopoEl+M6BdEfwnCJZtKxi1KgxUyJq13dy" crossorigin="anonymous">
+
+    <!-- Custom styles for this template -->
+    {% block customCSS %}
+    <link rel="stylesheet" href="stylesheets/style.css">
+    {% endblock %}
+  </head>
+
+  <body>
+
+      <!-- Temporary top navigation bar until permanent one is implemented -->
+      {% block navbar %}
+        <nav class="navbar navbar-fixed-top navbar-expand-md navbar-dark bg-dark mb-4">
+        <div class="container-fluid">
+          <div class="navbar-header">
+            <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation">
+                <span class="navbar-toggler-icon"></span>
+            </button>
+            <a class="navbar-brand" href="#">LUCCA (config. brand?)</a>
+          </div> <!--navbar-header-->
+
+          <!-- Collect the nav links and content for toggling -->
+          <!-- Each navbar button is its own block so child-pages can denote which is "active" -->
+          <div class="collapse navbar-collapse" id="navbarCollapse">
+            <ul class="navbar-nav mr-auto">
+              {% block Home %}
```

Comment:
Was wondering what is the purpose of surrounding nav bar items with blocks?

Reply:
The thought was to allow children pages to be able to signal to the user which navbar link was currently active. For example, have a look at the adminLogin.njk page. It replaces the adminLogin block with...:
```nunjucks
{% block AdminLogin %}
    <li class="nav-item active">
    <a class="nav-link" href="#">Admin Login<span class="sr-only">(current)</span></a>
    </li>
{% endblock %}
```
...to produce this:
![image](https://user-images.githubusercontent.com/33047755/35316973-42503f74-0089-11e8-9cc3-5fdb5e6e93dc.png)

Each page could do the same (e.g. Home.njk could add an active "Home" block without having to worry about the rest of the navbar to produce the following) which would give the user another visual cue of where they are on the website.
![image](https://user-images.githubusercontent.com/33047755/35317184-639dda82-008a-11e8-99d5-6e4d116ef3f3.png)

Reply:
Ah ok I see cool